Clarify default and internal namespace description (#1023)
temporalio/documentation: docs/concepts/what-is-a-namespace.md

You can use Namespaces to match the development lifecycle; for example, having separate `dev` and `prod` Namespaces.
Or you could use them to ensure Workflow Executions between different teams never communicate; such as ensuring that the `teamA` Namespace never impacts the `teamB` Namespace.

> - The Temporal Cluster uses the Namespace, “default”, if no other Namespace is specified, for all Temporal SDKs and the tctl.

Reviewer: ```suggestion
- If no other Namespace is specified, the Temporal Cluster uses the Namespace "default" for all Temporal SDKs and tctl.
```

>   - If you are deploying through Docker Compose or using the [auto-setup image](https://github.com/temporalio/docker-builds/blob/main/docker/auto-setup.sh) in a custom Docker Compose application, the Namespace, "default", is created, through the auto-setup script.

Reviewer: ```suggestion
  - If you are deploying through Docker Compose or using the [auto-setup image](https://github.com/temporalio/docker-builds/blob/main/docker/auto-setup.sh) in a custom Docker Compose application, the Namespace "default" is created, through the auto-setup script.
```

>   - If you are deploying through the [Temporal Helm charts](https://github.com/temporalio/helm-charts), you can create the "default" Namespace through the tctl, for example: `tctl namespace default`.

Reviewer: ```suggestion
  - If you are deploying through the [Temporal Helm charts](https://github.com/temporalio/helm-charts), you can create the "default" Namespace by using tctl; for example, `tctl namespace default`.
```

    We recommend using the default Namespace if you aren’t using multiple Namespaces.

> - **Case Insensitive**: Because of the DNS, namespaces are case-insensitive on the network/routing part.

Reviewer: ```suggestion
- **Case Insensitive**: Because of DNS, Namespaces are case insensitive on the network and routing side.
```
I also recommend lowercasing "Insensitive" in the run-in heading. If you do, please also lowercase "Configuration" in the "Namespace Configuration" run-in heading at the bottom of this list.

>   We recommend using lower case for namespace names to avoid potential issues.

Reviewer: ```suggestion
  We recommend using lowercase for Namespace names to avoid potential issues.
```
